app/src/pages/header/components/AddMovieForm.jsx

```diff
@@ -18,25 +20,40 @@ const ColoredButtonWrapper = styled(ColoredButton)`
 `
 
 function AddMovieForm(props) {
-    const [genres, setGenres] = useState([]);
-    
-    useEffect(() => {
-        Promise
-            .resolve(genresFromJson)
-            .then(genres => setGenres(genres))
-    }, [])
+    const [title, setTitle] = useState('');
+    const [releaseDate, setReleaseDate] = useState('');
+    const [url, setUrl] = useState('');
+    const [genres, setGenres] = useState('');
+    const [overview, setOverview] = useState('');
+    const [runtime, setRuntime] = useState(0);
+
+    function confirmAdd(event){
+        event.preventDefault();
+        var movie = {
+            title: title,
+            release_date: releaseDate,
+            poster_path: url,
+            genres: genres.split(','),
+            overview: overview,
+            runtime: Number.parseInt(runtime)
+        }
+
+        store.dispatch(addMovie(movie));

```

Comment:
u should use mapDispatchToProps and connect / or useDispatch hook

Reply:
connect(mapStateToProps, mapDispatchToProps)(AddMovieForm)

Reply:
Done: used `connect` from `react-redux`